docker/base/cleanup.sh

```diff
@@ -1,66 +1,85 @@
 #!/bin/bash
 #
 # Cleanup script for Firefly app
+#
 
-# /usr/bin/find if testing locally
+DEFAULT_CLEANUP_INTERVAL=2d
 find_cmd="/usr/bin/find"
-echo "cleanup: started"
-
-function doCleanup() {
-    workarea="${1}"
+re='^[0-9]+$'
 
-    # find cleanup logs less than 23 hours old
-    recent_cleanup=`${find_cmd} "${workarea}/cleanup_logs" -type f -mmin -$((60*23))`
+if [ -z "${CLEANUP_INTERVAL}" ]; then
+   CLEANUP_INTERVAL=${DEFAULT_CLEANUP_INTERVAL}
+fi
 
-    if [[ -z "${recent_cleanup}" ]] ; then
+function getMultiplier() { #return the minute multiplier the unit (h or m or d), 24h would return 60
+    instr="${1}"
+    u=$(echo "${instr}" | cut -c"${#instr}"-"${#instr}")
+    if [  "${u}" == 'm' ]; then echo 1;
+    elif [ "${u}" == 'd'  ]; then echo 1440
+    elif [ "${u}" == 'h'  ]; then echo 60
+    else echo 0
+    fi
+}
 
-        timestamp=$(date +20%y%m%dT%H%M%S)
-        log_file="${workarea}/cleanup_logs/cleanup.${timestamp}.log"
+function getInterval() { # get the interval without the unit i.e. "24h" would return a 24
+    instr="${1}"
+    echo ${instr} | cut -c-$((${#instr}-1))
+}
 
-        echo "Cleaning up old work files in ${workarea}" > "${log_file}" 2>&1
-        ${find_cmd} "${workarea}/stage" -type f -mtime +7 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        # master catalogs should be cleared more often than other files
-        ${find_cmd} "${workarea}/perm_files" -name 'master*' -mtime +1 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        ${find_cmd} "${workarea}/perm_files" -type f -mtime +7 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        ${find_cmd} "${workarea}/temp_files" -type f -mtime +1 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        ${find_cmd} "${workarea}/visualize/fits-cache" -type f -mtime +7 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        if [ -d "${workarea}/visualize/users" ]; then
-            ${find_cmd} "${workarea}/visualize/users" -type f -mtime +1 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        fi
+interval=$(getInterval "${CLEANUP_INTERVAL}")
+mult=$(getMultiplier "${CLEANUP_INTERVAL}")
+if  ! [[ "${interval}" =~ $re ]] || [ "$mult" -eq 0 ]; then  # check for an invalid CLEANUP_INTERVAL
+   echo "CLEANUP_INTERVAL of ${CLEANUP_INTERVAL} is invalid, must be number and m (minutes), h (hours) or d (days)"
+   echo '     Examples: 4h or 720m or 1d or 5d'
+   echo "     Setting CLEANUP_INTERVAL to 2 days (2d)"
+   CLEANUP_INTERVAL=${DEFAULT_CLEANUP_INTERVAL}
+   interval=$(getInterval "${DEFAULT_CLEANUP_INTERVAL}")
+   mult=$(getMultiplier "${DEFAULT_CLEANUP_INTERVAL}")
+fi
 
-        ${find_cmd} "${workarea}/upload" -type f -mtime +7 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
-        ${find_cmd} "${workarea}/HiPS" -type f -mtime +90 -exec /bin/rm '{}' \+ -print >> "${log_file}" 2>&1
+cleanupMinutes=$((interval*mult))
+sleepTime="$((cleanupMinutes+1))m"
+logfileMaxAge=$((cleanupMinutes*10+5))
```

Comment:
I am not sure why the +1m to sleepTime.  This makes it possible for a file to be older than cleanupMinutes.  Also, logfileMaxAge is not needed.  The size of the log file is insignificant.  It's served for debugging purposes mainly.  It should be set to a fixed time.  But, both are trivial.  Ok if you want to merge as is.

Reply:
Add +1 to sleep time allows it get search the entire interval. I think added the +1 make it possible to catch more older files. Otherwise it could miss something that we within a few seconds. It is not a big deal with 24h stuff my since LSST wants to set it to like 1h it might be more important.

Every cleanup run has a new log file. The max again keeps it to around 10 files.  It was hardcoded before.